temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java

```diff
@@ -1181,6 +1183,19 @@ public static Logger getLogger(String name) {
     return WorkflowInternal.getLastCompletionResult(resultClass, resultClass);
   }
 
+  /**
+   * Extract the latest failure from some previous run for this cron workflow. This is used in
+   * combination with cron schedule. A workflow can be started with an optional cron schedule. If
+   * any previous workflow has failed, this function returns that failure. If no previous workflows
+   * have failed, an empty optional is returned.
+   *
+   * @return The last {@link Exception} that occurred in this workflow, if there has been one.
+   */
+  public static Optional<Exception> getLastFailure() {
+    return WorkflowInternal.getLastFailure()
+        .map(f -> FailureConverter.failureToException(f, DataConverter.getDefaultInstance()));
```

Comment:
I would move this logic into WorkflowInternal. The idea is that Workflow is just a public layer over internal stuff without any business logic.